### PR TITLE
User story 4

### PR DESCRIPTION
**Pull request by EricLarson2020**
Attach github api to application and displayed github repos  on the dashboard

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,5 +1,22 @@
 class UsersController < ApplicationController
-  def show; end
```
Comment (EricLarson2020) [inline app/controllers/users_controller.rb:2]: Move this to another class and break up into separate methods

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -10,4 +10,14 @@
   <section>
     <h1>Bookmarked Segments</h1>
   </section>
+
+</section>
+</br>
+<section class="github"> <h2>Github Repos</h2>
+  <ul>
+    <% if !@links.nil? %>
```
Comment (EricLarson2020) [inline app/views/users/show.html.erb:18]: Nice use of logic here

File: spec/features/user/user_can_see_github_repositories_spec.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require "rails_helper"
+
+RSpec.describe 'User' do
```
Comment (EricLarson2020) [inline spec/features/user/user_can_see_github_repositories_spec.rb:3]: We should make the test more verbose

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,5 +1,22 @@
...
+    response = conn.get("user/repos")
+
+    parsed = JSON.parse(response.body, symbolized_names: true)[0..4]
+
+    @links = Hash.new(0)
+      parsed.map do |entry|
+      @links[entry["name"]] = entry["url"]
+    end
+  end
+
```
Comment (Lithnotep) [inline app/controllers/users_controller.rb:17]: We can come back a refactor this soon!

File: db/migrate/20200701212803_add_token_to_users.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddTokenToUsers < ActiveRecord::Migration[5.2]
+  def change
+    add_column :users, :token, :string
+  end
```
Comment (Lithnotep) [inline db/migrate/20200701212803_add_token_to_users.rb:4]: nice! we may need to rename this if we end up with more then one token type per user.

**Review by Lithnotep — APPROVED** [review]

**Pull request merged**